packages/effect/src/Utils.ts

```diff
@@ -784,3 +784,23 @@ export const structuralRegion = <A>(body: () => A, tester?: (a: unknown, b: unkn
     structuralRegionState.tester = currentTester
   }
 }
+
+const tracingFunction = (name: string) => {
+  const internalCall = <A>(body: () => A): A => body()
```

Comment:
Maybe we can use an object to prevent the rename perf hit:

```ts
const obj = {
  [name]() { return body() }
}
return obj[name]
```

Reply:
Rename is done once at module load, not at function call so not sure the cost has an impact. The issue with the object method is that apparently it isn't preserved in some cases, I.e. in the playground it doesn't work (check the repro)